prow/cmd/pipeline/controller.go

```diff
@@ -350,11 +350,12 @@ func (c *controller) createPipelineRun(pContext, namespace string, p *pipelinev1
 		return p, err
 	}
 	// Block until the pipelinerun is in the lister, otherwise we may attempt to create it again
-	err = wait.Poll(time.Second, 3*time.Second, func() (bool, error) {
```

Comment:
Can you elaborate more on what you are trying to achieve here? It seems that your changes don't change the affected result at all.

Reply:
But we want to exit if we get an error. Is there any case were we need to handle specific errors?

Reply:
the getPipelineRun function is a function aligning single responsibility principle. the function of polling should be from the caller, which is also called reconciler.